test/installer.js

```diff
@@ -265,4 +265,85 @@ describe('module', function () {
       })
     })
   })
+
+  describe('with duplicate dependencies', function (test) {
+    var dest = 'test/fixtures/out/kjfq/'
+
+    // User options with duplicates (including defaults duplicates)
+    var depends = ['libnss3', 'libxtst6', 'dbus', 'dbus']
+    var recommends = ['pulseaudio | libasound2', 'bzip2', 'bzip2']
+    var suggests = ['lsb-release', 'gvfs', 'gvfs']
+    var enhances = ['libc6', 'libc6']
+    var preDepends = ['footest', 'footest']
+
+    before(function (done) {
+      installer({
+        src: 'test/fixtures/app-with-asar/',
+        dest: dest,
+        rename: function (dest) {
+          return path.join(dest, '<%= name %>_<%= arch %>.deb')
+        },
+
+        options: {
+          productDescription: 'Just a test.',
+          section: 'devel',
+          priority: 'optional',
+          arch: 'i386',
+          depends: depends,
+          recommends: recommends,
+          suggests: suggests,
+          enhances: enhances,
+          preDepends: preDepends,
+          categories: []
+        }
+      }, done)
+    })
+
+    after(function (done) {
+      fs.remove(dest, done)
+    })
+
+    it('removes duplicate dependencies', function (done) {
+      access(dest + 'footest_i386.deb', function () {
+        child.exec('dpkg-deb -f footest_i386.deb ' +
+          'Depends Recommends Suggests Enhances Pre-Depends', { cwd: dest }, function (err, stdout, stderr) {
+          if (err) return done(err)
+          if (stderr) return done(new Error(stderr.toString()))
+
+          var baseDependencies = {
+            Depends: _.sortBy(_.union([
+              'gvfs-bin',
+              'libgconf2-4',
+              'libgtk2.0-0',
+              'libnotify4',
+              'libnss3',
+              'libxtst6',
+              'xdg-utils'
+            ], depends)),
+            Recommends: _.sortBy(_.union(['pulseaudio | libasound2'], recommends)),
+            Suggests: _.sortBy(_.union([
+              'gir1.2-gnomekeyring-1.0',
+              'libgnome-keyring0',
+              'lsb-release'
+            ], suggests)),
+            Enhances: _.sortBy(_.uniq(enhances)),
+            'Pre-Depends': _.sortBy(_.uniq(preDepends))
+          } // object with both user and default dependencies based on src/installer.js
+
+          // Creates object based on stdout (values are still strings)
+          var destDependecies = _.fromPairs(_.chunk(_.initial(stdout.split(/\n|:\s/)), 2))
+          // String values are mapped into sorted arrays
+          destDependecies = _.mapValues(destDependecies, function (value) {
+            if (value) return _.sortBy(value.split(', '))
+          })
+
+          if (_.isEqual(baseDependencies, destDependecies)) {
+            done()
+          } else {
+            done(new Error('There are duplicate dependencies'))
+          }
+        })
+      })
+    })
+  })
```

Comment:
We need to start deduplicating the tests...

Reply:
I agree. The first two can be eliminated since they are covered by the other tests.
Perhaps the test about Debian scripts can include other options since that test includes lintian checks, which covers various tests in itself.
At the very least we could just divide the test into multiple files to improve readability.
Nonetheless, I think those changes should be in a different PR.